src/core_tree_utils.c

```diff
@@ -128,7 +128,7 @@ int reorder_lhalo_to_lhvt(const int32_t nhalos, struct halo_data *forest, int32_
                     return EXIT_FAILURE;
                 }
             } else {
-                XASSERT(desc >= 0 && desc < nhalos, -1,
+                XRETURN(desc >= 0 && desc < nhalos, -1,
```

Comment:
In the files down below, you're returning explicitly `-EXIT_FAILURE`.  Can these returns of `-1` be changed for consistency?

Reply:
Looks like the return value could be `EXIT_FAILURE` and does not need to be negative. I have updated accordingly